### ISSUE TITLE
Is there any problems in preload?

### ISSUE THREAD

**Issue by mikashin** [processing/p5.js#1481]
Hi!
I used preload() to load my table in PC. But it didn't work.
So I tried the example code in your website (http://p5js.org/reference/#/p5/loadTable ) and found that preload() doesn't work. It caused  "22462: Uncaught TypeError: object is not a function"  error.  
So I ran a script from the example code as below.

```javascript
var table;

//function preload(){
// table=loadTable("http://p5js.org/reference/assets/mammals.csv", "csv", "header");
//}

function setup() {
  a=1;
  print(a);
  table=loadTable("http://p5js.org/reference/assets/mammals.csv", "csv", "header");
  print(table.getColumn("name"));
  print("hi");
  
  for (var r = 0; r < table.getRowCount(); r++)
    for (var c = 0; c < table.getColumnCount(); c++) {
      print(table.getString(r, c));
    }
  
 
}
```

The outcome was
1
[]
"hi"

As you see, the example data didn't come out.
I think preload() and loading components in table doesn't work properly. Could you confirm this?

**Comment by lmccart** [comment]
Preload seems to be working for me. The code above as printed gives the output because you are trying to print parts of the table before the table has fully loaded. This is the code that I'm using. I'm working locally and have downloaded the mammals.csv file locally as well and placed in a folder called assets.

```javascript
var table;

function preload(){
  table=loadTable("assets/mammals.csv", "csv", "header");
}

function setup() {
  print(table.getColumn("name"));
  for (var r = 0; r < table.getRowCount(); r++)
    for (var c = 0; c < table.getColumnCount(); c++) {
      print(table.getString(r, c));
    }
}
```

outputs:
```
["Goat", "Leopard", "Zebra"]
0
Capra hircus
Goat
1
Panthera pardus
Leopard
2
Equus zebra
Zebra
```

**Comment by toolness** [comment]
@mikashin, one thing I've noticed about your code is that it's attempting to retrieve http://p5js.org/reference/assets/mammals.csv, which will fail because p5js.org isn't serving those files with [Cross-Origin Resource Sharing (CORS)](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing) headers. This is part of why @lmccart downloaded the file locally and her code retrieves it via a relative URL... It's probably also why your original attempt at using `loadTable()` within `preload()` didn't work.

@lmccart, do you think `loadTable()` and similar functions should log warnings or throw errors in these sorts of situations when they're called outside of `preload()` but not passed a callback? Also curious if you think p5js.org should serve `/reference/assets/` with CORS headers so that @mikashin's original use of `preload()` might have worked.

**Comment by toolness** [comment]
@mikashin, actually, the `22462: Uncaught TypeError: object is not a function` error you received is indeed a bug in p5! I've attempted to fix it in #1500. Thanks for reporting it!

**Comment by lmccart** [comment]
fixed with https://github.com/processing/p5.js/pull/1500
